### PR TITLE
Improve json schema validation messages

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Improve json schema validation messages

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12694

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/validator/DevfileSchemaValidator.java [hunk 1]
```diff
@@ -55,14 +55,41 @@ public JsonNode validateBySchema(String yamlContent, boolean verbose)
...
-              .map(message -> verbose ? message.asJson().toString() : message.getMessage())
-              .collect(Collectors.joining(", ", "[", "]"));
-      throw new DevfileFormatException(
-          format("Devfile schema validation failed. Errors: %s", error));
+      throw new DevfileFormatException(prepareErrorMessage(report));
     }
     return data;
   }
+
+  private String prepareErrorMessage(ProcessingReport report) {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/validator/DevfileSchemaValidator.java]: Could you please move it in a separate class, add detailed java doc, and some unit tests

File: wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/validator/DevfileSchemaValidatorTest.java [hunk 2]
```diff
@@ -66,60 +66,60 @@ public void shouldThrowExceptionOnValidationNonValidDevfile(
...
       {
         "command/devfile_multiple_commands_actions.yaml",
-        "Devfile schema validation failed. Errors: \\[array is too long: must have at most 1 elements but instance has 2 elements\\]$"
+        "Devfile schema validation failed. Error: /devfile/commands/0/actions array is too long: must have at most 1 elements but instance has 2 elements$"
       },
       // cheEditor/chePlugin tool model testing
       {
         "editor_plugin_tool/devfile_editor_tool_with_missing_id.yaml",
-        "Devfile schema validation failed\\. Errors: \\[instance failed to match exactly one schema \\(matched 0 out of 3\\)\\]"
+        "Devfile schema validation failed\\. Errors: \\[instance failed to match exactly one schema \\(matched 0 out of 3\\),/devfile/tools/0 object has missing required properties \\(\\[\"id\"\\]\\),/devfile/tools/0 object has missing required properties \\(\\[\"local\"\\]\\),/devfile/tools/0 object has missing required properties \\(\\[\"image\",\"memoryLimit\"\\]\\)\\]"
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/validator/DevfileSchemaValidatorTest.java:118]: It definitely looks much detailed than before but It still does not look clear =(
I think `if`/`else` statement is the only thing that can help us to provide pretty good UX

**Review by sleshchenko — APPROVED** [review]
The draft looks good to me :+1: 
Keep improving it.

**Comment by mshaposhnik** [comment]
ci-build

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by mshaposhnik** [comment]
ci-test

**Review by skabashnyuk — APPROVED** [review]

**Comment by garagatyi** [comment]
Nice refactoring :+1:

**Comment by artaleks9** [comment]
Success Rate = 100%
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1572/) doesn't show any regression against this Pull Request.

**Pull request merged**